packages/trader/src/AppV2/Hooks/useContractsForCompany.ts

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+import { useTraderStore } from 'Stores/useTraderStores';
+import { useStore } from '@deriv/stores';
+import { cloneObject, getContractCategoriesConfig, getContractTypesConfig, WS } from '@deriv/shared';
+
+type TContractType = {
+    text?: string;
+    value: string;
+};
+
+export type TContractTypesList = {
+    [key: string]: {
+        name: string;
+        categories: DeepRequired<TContractType[]>;
+    };
+};
+
+const useContractsForCompany = () => {
+    const [contract_types_list, setContractTypesList] = React.useState<TContractTypesList | []>([]);
+    const { setContractTypesListV2 } = useTraderStore();
+    const { client } = useStore();
+    const { landing_company_shortcode } = client;
+    const contract_categories = getContractCategoriesConfig();
+    const available_categories = cloneObject(contract_categories);
+    const contract_types = getContractTypesConfig();
+    let available_contract_types: ReturnType<typeof getContractTypesConfig> = {};
+
+    const fetchContractForCompany = React.useCallback(async () => {
+        let response;
+
+        const request = {
+            landing_company: landing_company_shortcode,
+        };
+
+        try {
+            response = await WS.contractsForCompany(request);
```

Comment:
the request and responses are the same when user is not logged in?

Reply:
does the backend return a standard response when landing_company = '' ? since we dont have landing_company when not logged in

Reply:
the reason why im asking is we need response from this to call active_symbols and this call is needed to render available trade types (when logged in or not)

Reply:
The default landing company is virtual, so when users are logged out and we don't provide any landing company, BE defaults it to virtual

Reply:
I think it's the same as what we have on production, disregarding your landing_company, it shows all the trade types once you're logged out